js/foam/node/handlers/FileHandler.js

```diff
@@ -47,10 +57,23 @@ CLASS({
       if ( this.url.parse(req.url).pathname !== this.pathname ) return false;
 
       this.fs.readFile(this.file, function(err, data) {
```

Comment:
Might be better to restructure this as

this.fs.createReadStream(this.file) similar to the StaticFileHandler